### ISSUE TITLE
Triple Press and quadruple press option for aq2 switch in Homekit

### ISSUE THREAD

**Issue by Krocko** [ebaauw/homebridge-hue#333]
Hi Erik. Can you add triple press and quadruple press (buttonevent 1005 and 1006) from the Xiaomi Aqara Smart Switch to HomeKit?

**Comment by ebaauw** [comment]
Triple Press and Quadruple Press (and Shake) are already supported (as _Double Press_).  I'm happy to change Triple Press and Quadruple Press to _Long Press_.  Alternatively, I could expose a custom characteristic with the `buttonevent` value, so you can create your own HomeKit rules, but this wouldn't be supported by Apple's Home app.

**Background**

HomeKit only supports actions for _Single Press_ (i.e. press/release), _Double Press_ (i.e. press/release/press/release), and _Long Press_ (i.e. press/hold/release).  There's nothing I can do to change that.  homebridge-hue maps the values for `state.buttonevent` to these HomeKit actions:

buttonevent | value | HomeKit action
------------ | ------ | ---------------
Press | x000 | (ignored)
Hold | x001 | _Long Press_ (only first event)
Short Release | x002 | _Short Press_
Long Release | x003 | _Long Press_ (only first event, when x001 was missed)
Double press* | x004 | _Double Press_
Triple press* | x005 | _Double Press_
Quadruple press* | x006 |  _Double Press_
Shake* | x007 |  _Double Press_

*) deCONZ only.

Note that homebridge-hue needs to poll the Hue bridge (as it doesn't support notifications like deCONZ), so it doesn't reliably receive all buttonevent changes.  In particular, the press is almost always missed in a press/short release sequence as are the press and hold in a press/hold/long release sequence.

**Comment by Krocko** [comment]
OK, so i can only set an action for short press and an action for double press in HomeKit?

**Comment by ebaauw** [comment]
Currently yes.  I could add an action for _Long Press_, mapped to triple and quadruple press.

**Comment by Krocko** [comment]
What would be nice.

**Comment by ebaauw** [comment]
In v0.9.1.

You can check the mapping in the log when running `homebridge -D`:
```
[2018-4-21 12:22:22] [Hue] Downstairs Switch: state changed event
[2018-4-21 12:22:22] [Hue] Downstairs Switch: sensor buttonevent 1002 on 2018-04-21T10:21:39
[2018-4-21 12:22:22] [Hue] Downstairs Switch Button: homekit button single press
[2018-4-21 12:22:24] [Hue] Downstairs Switch: state changed event
[2018-4-21 12:22:24] [Hue] Downstairs Switch: sensor buttonevent 1004 on 2018-04-21T10:22:22
[2018-4-21 12:22:24] [Hue] Downstairs Switch Button: homekit button double press
[2018-4-21 12:22:26] [Hue] Downstairs Switch: state changed event
[2018-4-21 12:22:26] [Hue] Downstairs Switch: sensor buttonevent 1005 on 2018-04-21T10:22:24
[2018-4-21 12:22:26] [Hue] Downstairs Switch Button: homekit button long press
[2018-4-21 12:22:28] [Hue] Downstairs Switch: state changed event
[2018-4-21 12:22:28] [Hue] Downstairs Switch: sensor buttonevent 1006 on 2018-04-21T10:22:26
[2018-4-21 12:22:28] [Hue] Downstairs Switch Button: homekit button long press
```

**Comment by ebaauw** [comment]
Hm, I changed the buttonevent handler, but forgot to change the allowed values for the _Programmable Switch Event_ characteristic.  Fixed in v0.9.2.

**Comment by Krocko** [comment]
It is working now. Thank you.
